docker/docker_image_src.go

```diff
@@ -65,11 +73,37 @@ func (s *dockerImageSource) GetManifest() (manifest types.ImageManifest, unverif
 	if res.StatusCode != http.StatusOK {
 		return nil, "", errFetchManifest{res.StatusCode, manblob}
 	}
+	// TODO(remove, already set in manifest, below): Miloslav?
```

Comment:
> however my current understanding is that this digest is not to be trusted so I don't think it's needed to be exposed in skopeo - correct?

The model in presence of signature assumes the registry to be malicious: it can serve a completely different manifest from the one which has been signed, and it can lie about the digest (e.g. when asked for a manifest with a specific digest, it can return a manifest with a different digest value; and whatever manifest it returns, it can return a non-matching value in the HTTP header).

_If_ we don’t trust the registry and want to verify signatures, when verifying signatures we _must_ recompute the digest from the blob we have actually received.

OTOH if we are displaying information from the registry without doing any signature verification, we are ~by definition trusting the registry and using the value would be fine.

Still, there seems to be no reason to show the possibly untrusted value to the user: `skopeo inspect` can recompute a gurananteed-correct value using `dockerutils.ManifestDigest` and we can simply ignore the HTTP header.

AFAICT the only thing the HTTP header could be useful for is protection against data corruption with non-TLS HTTP with only the 16-bit TCP checksums, or against using HTTP incorrectly, causing the data to be corrupted during transfer; this would benefit from the uploader and downloader verifying the same digest (regardless of whether higher layers know about any signatures). We don't currently do this; and even if we did, the digest in the HTTP header would be used only by the `ImageSource` / `ImageDestination` transport and there isn’t really a reason to provide it to callers.